### PR TITLE
Prevent panic in `api-key list` and `describe`

### PR DESCRIPTION
**Pull request by sgagniere**
Release Notes
-------------

Breaking Changes
- PLACEHOLDER

New Features
- PLACEHOLDER

Bug Fixes
- Prevent a panic which could occur in various ``confluent api-key`` commands

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
If `State.Auth.User` is `nil` for whatever reason, a panic will result in `getAllUsers` when that function attempts to access the `User.Id` field.

References
----------

Test & Review
-------------

Open Questions / Follow-ups
---------------------------
Followup: investigate the cause of incorrectly populated contexts.

### CODE REVIEW

File: internal/pkg/config/v1/context_state.go [hunk 1]
```diff
@@ -19,6 +21,20 @@ type ContextState struct {
 	Nonce            []byte      `json:"nonce,omitempty"`
 }
 
+func (c *ContextState) GetAuth() *AuthConfig {
```
Comment (brianstrauch) [inline internal/pkg/config/v1/context_state.go:24]: This function already exists for `Context`!

File: internal/pkg/config/v1/context_state.go [hunk 2]
```diff
@@ -19,6 +21,20 @@ type ContextState struct {
...
+	if c != nil {
+		return c.Auth
+	}
+	return nil
+}
+
+func (c *ContextState) GetUser() *ccloudv1.User {
```
Comment (brianstrauch) [inline internal/pkg/config/v1/context_state.go:31]: Same with this one!

File: internal/cmd/api-key/command_list.go [hunk 3]
```diff
@@ -153,7 +153,9 @@ func getServiceAccountsMap(serviceAccounts []iamv2.IamV2ServiceAccount) map[stri
 func getUsersMap(users []*ccloudv1.User) map[int32]*ccloudv1.User {
 	userMap := make(map[int32]*ccloudv1.User)
 	for _, user := range users {
-		userMap[user.Id] = user
+		if user != nil {
```
Comment (brianstrauch) [inline internal/cmd/api-key/command_list.go:156]: This check seems redundant given the other check made in this PR.

Comment (brianstrauch) [reply internal/cmd/api-key/command_list.go:156]: I'd rather use the getter syntax than adding defensive `if` statements everywhere.

File: internal/pkg/config/v1/context_state.go (hunk 1 above)
Comment (sgagniere) [reply internal/pkg/config/v1/context_state.go:24]: Oh right, I'll switch it around.

File: internal/cmd/api-key/command_list.go (hunk 3 above)
Comment (sgagniere) [reply internal/cmd/api-key/command_list.go:156]: It's mostly redundant, but it's to preempt another edge case: if the backend returns a `nil` user as part of the list.

Comment (sgagniere) [reply internal/cmd/api-key/command_list.go]: Ok I removed the if. It still has the getter function.

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**